### ISSUE TITLE
`CS0169` is reported only on build

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#63164]
Port of https://github.com/dotnet/razor-tooling/issues/6617

**Version Used**:
VS 2022 v17.3 Preview 6

**Steps to Reproduce**:
1) Create console app
2) Add this code:
```cs
Console.WriteLine("Hello world!");

class MyClass
{
    
}
```
3) Inside `MyClass` add any private field;

**Expected Behavior**:
`CS0169` appears in the error list as soon as I finish writing the statement. Note, that there is an `IDE0051` that does the same thing (duplicate diagnostic for the same thing?) and it appears almost immediatelly.

**Actual Behavior**:
`CS0169` is updated only on build. So if I build the project/solution it will appear in the list and is not gone even if I remove the field

**Demo:**
https://user-images.githubusercontent.com/70431552/182611087-80fe94b1-aea3-49e3-b7b7-9aca4fa0d6da.mp4

**Comment by mavasani** [comment]
This is by design, and https://github.com/dotnet/docs/issues/26708 tracks adding documentation for the same.
